test/linter/test-status-inheritance.js

```diff
@@ -0,0 +1,60 @@
+const chalk = require('chalk');
+const { Logger } = require('../utils.js');
+
+/**
+ * @typedef {import('../../types').Identifier} Identifier
+ * @typedef {import('../../types.js').StatusBlock} StatusBlock
+ */
+
+/**
+ * @param {Identifier} data
+ * @param {Logger} logger
+ * @param {StatusBlock?} parentStatus,
+ * @param {string[]} path
+ */
+function checkStatusInheritance(data, logger, parentStatus = {}, path = []) {
```

Comment:
@saschanaz do you want to add a "Unexpected standards track status while the parent is non-standards track"?